client/src/redux/actionCreators/location.ts

```diff
@@ -52,6 +52,65 @@ export const getLocationCurrWeather = (query: string) => async (dispatch: Dispat
   }
 };
 
+export const getLocationHourlyWeather = (query: string) => async (dispatch: Dispatch<Action>) => {
+  dispatch({ type: ActionType.GET_LOCATION_HOURLY_WEATHER_PENDING });
+
+  try {
+    const data = await axios.get(
+      `${BASE_URL}/forecaApi/forecast/hourly/${query}&tempunit=C&windunit=MS`,
+      { withCredentials: true }
+    );
+
+    dispatch({ type: ActionType.GET_LOCATION_HOURLY_WEATHER_SUCCESS, payload: data.data.forecast });
+  } catch (err) {
+    if (err instanceof Error) {
+      dispatch({ type: ActionType.GET_LOCATION_HOURLY_WEATHER_FAIL, payload: err.message });
+    }
+  }
+};
+
+export const getLocationThreeHourlyWeather = (query: string) => async (
+  dispatch: Dispatch<Action>
+) => {
+  dispatch({ type: ActionType.GET_LOCATION_THREE_HOURLY_WEATHER_PENDING });
+
+  try {
+    const data = await axios.get(
+      `${BASE_URL}/forecaApi/forecast/3hourly/${query}&tempunit=C&windunit=MS`,
+      { withCredentials: true }
+    );
+
+    dispatch({
+      type: ActionType.GET_LOCATION_THREE_HOURLY_WEATHER_SUCCESS,
+      payload: data.data.forecast
+    });
+  } catch (err) {
+    if (err instanceof Error) {
+      dispatch({ type: ActionType.GET_LOCATION_THREE_HOURLY_WEATHER_FAIL, payload: err.message });
+    }
+  }
+};
+
+export const getLocationDailyWeather = (query: string) => async (dispatch: Dispatch<Action>) => {
+  dispatch({ type: ActionType.GET_LOCATION_DAILY_WEATHER_PENDING });
+
+  try {
+    const data = await axios.get(
+      `${BASE_URL}/forecaApi/forecast/daily/${query}&tempunit=C&windunit=MS`,
+      { withCredentials: true }
+    );
+
+    dispatch({
+      type: ActionType.GET_LOCATION_DAILY_WEATHER_SUCCESS,
+      payload: data.data.forecast
+    });
+  } catch (err) {
+    if (err instanceof Error) {
+      dispatch({ type: ActionType.GET_LOCATION_DAILY_WEATHER_FAIL, payload: err.message });
+    }
+  }
+};
```

Comment:
can these 3 functions be united to one or maybe one additional generic function could be created? i see that only url and action types are different